controller/controller.go

```diff
@@ -89,10 +89,17 @@ func (c *controller) Start() error {
 		return errors.New("can't restart controller")
 	}
 
+	var startedUpdaters []Updater
 	for _, u := range c.updaters {
 		if err := u.Start(); err != nil {
+			for _, started := range startedUpdaters {
+				if err := started.Stop(); err != nil {
+					log.Warnf("unable to stop %s: %v", u, err)
+				}
+			}
 			return fmt.Errorf("unable to start %v: %v", u, err)
 		}
+		startedUpdaters = append(startedUpdaters, u)
```

Comment:
This is just an additional safeguard to make sure we don't leave things dangling or in a broken state. See the test.